crates/runtime/src/tools/builtin/list_datasets.rs

```diff
@@ -124,6 +133,63 @@ pub async fn get_table_elements(
         .collect_vec()
 }
 
+/// To get tables from catalogs, we must make SQL query on `information_schema.tables` and use to find all tables.
+pub async fn get_catalog_elements(
+    rt: Arc<Runtime>,
+    _opt_include: Option<&[String]>,
+) -> Result<Vec<ListDatasetElement>, Box<dyn std::error::Error + Send + Sync>> {
+    let Some(ref app) = *rt.app.read().await else {
+        return Ok(vec![]);
+    };
+
+    let catalogs: Vec<_> = app.catalogs.iter().map(|c| c.name.clone()).collect();
+
+    let batches = rt.datafusion().query_builder(format!(
+        "SELECT table_catalog, table_schema, table_name FROM information_schema.tables WHERE table_schema != 'information_schema' AND table_catalog in ({});",
```

Comment:
we can just ask datafusion itself for the catalogs, no need for a SQL query

Reply:
i.e. https://github.com/spiceai/spiceai/blob/trunk/crates/runtime/src/datafusion.rs#L336

Reply:
Nice, I didn't know this.